Source/SwiftyJSON.swift

```diff
@@ -123,6 +123,42 @@ public struct JSON {
         }
         self.init(dictionary)
     }
+    
+    /**
+     Merges another JSON into this JSON, whereas primitive values which are not present in this JSON are getting added, 
+     present values getting overwritten, array values getting appended and nested JSONs getting merged the same way.
+ 
+     - parameter other: The JSON which gets merged into this JSON
+     */
+    public mutating func merge(_ other: JSON) {
+        if self.type == other.type {
+            switch self.type {
+            case .dictionary:
+                for (key, _) in other {
+                    self[key].merge(other[key])
+                }
+            case .array:
+                self = JSON(self.arrayValue + other.arrayValue)
+            default:
+                self = other
+            }
+        } else {
+            self = other
```

Comment:
> My point is that we should always make sure that errors are propagated back to the client and that we corrupt data without one of those errors.

In my opinion it is not an error, but a proper use case of this merging function. And by prohibiting merging two key-value-pairs that are of different types, we would rob users of this possibility. 

I love the idea of python, which is "we're all consenting adults here". Programmers mostly know what they are doing, and if there is overriding is happening because of type mismatch it is either on purpose or it is not SwiftyJSON's tasks to prohibit that by cutting functionality. 

> If someone did want to replace the entire JSON when merging with conflicting types, they could simply do:

This is not the same as what is happening in my implementation. Of course if you call `merge`on top level with two JSONs differing in type, you would overwrite the first JSON, but in every other case, just the nested fields in the JSONs are overwritten. 

For instance if you have an "id" field with an integer and for whatever reason merge a JSON with an "id" field which holds a string, just this field gets overwritten. 

> Can you clarify what you mean by

The main point being: In my opinion it is the desired and expected behavior of merging two JSONs, because JSONs are not type-safe.

But I have asked around here in the office, and got one camp being the same opinion as mine and another camp who is with you. :)

Here is my suggestion:

````swift
/**
     Merges another JSON into this JSON, whereas primitive values which are not present in this JSON are getting added, 
     present values getting overwritten, array values getting appended and nested JSONs getting merged the same way.
 
     - parameter other: The JSON which gets merged into this JSON
     - parameter strict: Enforce typesafe merging. Not matching types for one key will trigger an exception.
     */
    public mutating func merge(with other: JSON, strict: Bool = false) throws {
        if self.type == other.type {
            switch self.type {
            case .dictionary:
                for (key, _) in other {
                    try self[key].merge(with: other[key], strict: strict)
                }
            case .array:
                self = JSON(self.arrayValue + other.arrayValue)
            default:
                self = other
            }
        } else {
            if strict {
                throw JSONError.mergeDifferentTypes
            }
            self = other
        }
    }
````
Thank you for your time.